contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxMetrics.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.contrib.jmxmetrics;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.util.Properties;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class JmxMetrics {
+  private static final Logger logger = Logger.getLogger(JmxMetrics.class.getName());
+
+  private final Timer timer = new Timer();
+  private GroovyRunner runner;
+
+  /**
+   * Begins the core metric scraping and reporting loop on configured interval after parsing and
+   * binding the configured groovy script and establishing the {@link JmxClient} connection.
+   *
+   * @param config - {@link JmxConfig} with Groovy script path, JMX connection info, and metric
+   *     export options.
+   */
+  public void start(final JmxConfig config) {
+    JmxClient jmxClient;
+    try {
+      jmxClient = new JmxClient(config);
+    } catch (MalformedURLException e) {
+      throw new ConfigureError("Malformed serviceUrl: ", e);
+    }
+
+    runner = new GroovyRunner(config.groovyScript, jmxClient, new GroovyUtils(config));
+
+    timer.scheduleAtFixedRate(
+        wrapTimerTask(
+            new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  runner.run();
+                } catch (Throwable e) {
+                  logger.log(Level.SEVERE, "Error gathering JMX metrics", e);
+                }
+              }
+            }),
+        0,
+        config.intervalMilliseconds);
+    logger.info("Started GroovyRunner.");
+  }
+
+  private static TimerTask wrapTimerTask(final Runnable r) {
+    return new TimerTask() {
+      @Override
+      public void run() {
+        r.run();
+      }
+    };
+  }
+
+  private void shutdown() {
+    logger.info("Shutting down JmxMetrics Groovy runner and exporting final metrics.");
+    timer.cancel();
+    runner.flush();
+  }
+
+  private static JmxConfig getConfigFromArgs(final String[] args) {
+    if (args.length != 0 && (args.length != 2 || !args[0].equalsIgnoreCase("-config"))) {
+      System.out.println(
+          "Usage: java io.opentelemetry.contrib.jmxmetrics.JmxMetrics "
+              + "-config <path_to_config.properties>");
```

Comment:
In most of our configuration we support all three of environment variables, system properties, and config file. I think if we wanted to keep it simple for now, we should start with system properties to reduce the number of parts.

Reply:
I'll need to add properties file content via stdin next, and think it's better ux to accept a file than to require everything (including password) over the command line.  Are you ok w/ me leaving as is?  It's not a required option as is.